gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/vertx/view/thymeleaf/GraviteeMessageResolver.java

```diff
@@ -0,0 +1,102 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.thymeleaf.context.ITemplateContext;
+import org.thymeleaf.messageresolver.AbstractMessageResolver;
+import org.thymeleaf.messageresolver.IMessageResolver;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Optional.ofNullable;
+
+/**
+ * @author Eric LELEU (eric.leleu at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class GraviteeMessageResolver extends AbstractMessageResolver implements IMessageResolver {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    public static final String DEFAULT_LOCALE = "default";
+    Map<String, Properties> propertiesMap;
+
+    public GraviteeMessageResolver(String i18nLocation) {
+        final File directory = Paths.get(i18nLocation).toFile();
+        if (directory.exists() && directory.isDirectory()) {
+            this.propertiesMap = Stream.of(directory.listFiles())
+                    .filter(file -> !file.isDirectory())
+                    .filter(file -> file.getName().startsWith("messages"))
+                    .filter(file -> file.getName().endsWith(".properties"))
+                    .map(file -> {
+                        try (InputStream input = new FileInputStream(file)) {
+                            Properties prop = new Properties();
+                            prop.load(input);
+                            final int localIndex = file.getName().indexOf("_");
+                            final int suffixIndex = file.getName().indexOf(".");
+                            final String locale = localIndex > 0 ? new Locale(file.getName().substring(localIndex + 1, suffixIndex)).getLanguage() : DEFAULT_LOCALE;
+                            return Map.of(locale, prop);
+                        } catch (IOException e) {
+                            logger.debug("i18n File '{}' can't be loaded", file.getName(), e);
+                            return Map.<String, Properties>of();
+                        }
+                    })
+                    .map(Map::entrySet)
+                    .flatMap(Set::stream)
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        } else {
+            this.propertiesMap = new HashMap<>();
+        }
+
+        // create empty default properties to avoid NPE during message lookup
+        this.propertiesMap.putIfAbsent(DEFAULT_LOCALE, new Properties());
+
+        this.setOrder(0);
+    }
+
+    public boolean isSupported(Locale locale, boolean lax) {
+        return (!lax && this.propertiesMap.containsKey(locale.toString())) || (lax && this.propertiesMap.containsKey(locale.getLanguage()));
```

Comment:
I don't understand the purpose of the lax boolean.

Reply:
Indeed, it is useless now as I call this method with the `language.tag()` to get only the language code and not the raw value (fr-FR) provided by `language.value()`